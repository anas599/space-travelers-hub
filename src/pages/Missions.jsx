import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import SingleMission from '../components/SingleMission';
import { allMissions, getMissionFromAPI } from '../redux/mission/missionSlice';

const Missions = () => {
  const missions = useSelector(allMissions);
  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(getMissionFromAPI());
    }
  });
  return (
    <section className="missionContainer">
      <table className="missions">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <SingleMission key={mission.id} mission={mission} />
          ))}
        </tbody>
      </table>
    </section>
  );
};
export default Missions;
