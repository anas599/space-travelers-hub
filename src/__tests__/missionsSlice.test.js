import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Missions from '../pages/Missions';

const mockStore = configureStore([]);
const description = 'Mission 1 Desc';

describe('Missions Page', () => {
  it('single mission info should display', () => {
    const missionsArr = [
      {
        id: '1',
        missionName: 'Mission 1',
        description,
        joined: false,
      },
    ];
    const initialState = {
      missions: {
        missions: missionsArr,
      },
    };
    const store = mockStore(initialState);
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const missionName = screen.getByText('Mission 1');
    const missionDesc = screen.getByText(description);
    expect(missionName).toBeInTheDocument();
    expect(missionDesc).toBeInTheDocument();
    expect(store.getState()).toMatchSnapshot();
  });
});
