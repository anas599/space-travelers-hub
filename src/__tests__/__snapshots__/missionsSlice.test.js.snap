// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Page single mission info should display 1`] = `
Object {
  "missions": Object {
    "missions": Array [
      Object {
        "description": "Mission 1 Desc",
        "id": "1",
        "joined": false,
        "missionName": "Mission 1",
      },
    ],
  },
}
`;

exports[`Single mission info to display 1`] = `
Object {
  "missions": Object {
    "missions": Array [
      Object {
        "description": "Mission 1 description",
        "id": "1",
        "joined": false,
        "missionName": "Mission 1",
      },
    ],
  },
}
`;
